c_binding/lsm_local_disk.cpp

```diff
@@ -1428,3 +1430,242 @@ int lsm_local_disk_link_speed_get(const char *disk_path, uint32_t *link_speed,
+    }
+
+    if (led_scan(ctx) != LED_STATUS_SUCCESS) {
+        led_free(ctx);
+        return LSM_ERR_NO_MEMORY;
+    }
+
+    *handle = (lsm_led_handle *)calloc(1, sizeof(lsm_led_handle));
+    if (!(*handle)) {
+        return LSM_ERR_NO_MEMORY;
+    }
+    (*handle)->magic = LSM_LED_HANDLE_MAGIC;
+    (*handle)->ctx = ctx;
+    return LSM_ERR_OK;
+}
+
+void lsm_led_handle_free(lsm_led_handle *handle) {
+    if (LSM_IS_LED_HANDLE(handle)) {
+        led_free(handle->ctx);
+        handle->ctx = NULL;
+        handle->magic = 0;
+        free(handle);
+    }
+}
+
+int lsm_led_slot_iterator_get(lsm_led_handle *handle,
+                              lsm_led_slot_itr **slot_itr, lsm_error **lsm_err,
+                              lsm_flag flags) {
+
+    lsm_led_slot_itr *itr = NULL;
+    led_status_t led_rc = LED_STATUS_SUCCESS;
+
+    if (!LSM_IS_LED_HANDLE(handle) || CHECK_RP(slot_itr) || CHECK_RP(lsm_err) ||
+        LSM_FLAG_UNUSED_CHECK(flags)) {
+        return LSM_ERR_INVALID_ARGUMENT;
+    }
+
+    itr = (lsm_led_slot_itr *)calloc(1, sizeof(lsm_led_slot_itr));
+    if (!itr) {
+        return LSM_ERR_NO_MEMORY;
+    }
+
+    itr->magic = LSM_LED_SLOT_ITR_MAGIC;
+    itr->entry.magic = LSM_LED_SLOT_MAGIC;
+
+    led_rc = led_slots_get(handle->ctx, &itr->sl);
+    if (led_rc != LED_STATUS_SUCCESS) {
+        *lsm_err = translate_led_to_lsm(
+            led_rc, "Unexpected return for 'led_slots_get'");
+        return LSM_ERR_LIB_BUG;
+    }
+
+    *slot_itr = itr;
+    return LSM_ERR_OK;
+}
+
+void LSM_DLL_EXPORT lsm_led_slot_iterator_free(lsm_led_handle *handle,
+                                               lsm_led_slot_itr *slot_itr) {
+    if (LSM_IS_LED_HANDLE(handle) && LSM_IS_LED_SLOT_ITR(slot_itr)) {
+        slot_itr->entry.slot = NULL;
+        slot_itr->entry.magic = 0;
+        led_slot_list_free(slot_itr->sl);
+    }
+}
+
+void LSM_DLL_EXPORT lsm_led_slot_iterator_reset(lsm_led_handle *handle,
+                                                lsm_led_slot_itr *slot_itr) {
+    if (LSM_IS_LED_HANDLE(handle) && LSM_IS_LED_SLOT_ITR(slot_itr)) {
+        led_slot_list_reset(slot_itr->sl);
+    }
+}
+
+lsm_led_slot *lsm_led_slot_next(lsm_led_handle *handle,
+                                lsm_led_slot_itr *slot_itr) {
+    if (LSM_IS_LED_HANDLE(handle) && LSM_IS_LED_SLOT_ITR(slot_itr)) {
+        slot_itr->entry.slot = led_slot_next(slot_itr->sl);
+        if (slot_itr->entry.slot) {
+            return &slot_itr->entry;
+        }
+    }
+    return NULL;
+}
+
+const char LSM_DLL_EXPORT *lsm_led_slot_id(lsm_led_slot *slot) {
+    if (LSM_IS_LED_SLOT_ENTRY(slot)) {
+        return led_slot_id(slot->slot);
+    }
+    return NULL;
+}
+
+const char LSM_DLL_EXPORT *lsm_led_slot_device(lsm_led_slot *slot) {
+    if (LSM_IS_LED_SLOT_ENTRY(slot)) {
+        return led_slot_device(slot->slot);
+    }
+    return NULL;
+}
+
+uint32_t lsm_led_slot_status_get(lsm_led_slot *slot) {
+    uint32_t translated = LSM_DISK_LED_STATUS_UNKNOWN;
```

Comment:
would it make sense to call !LSM_IS_LED_SLOT_ENTRY(slot) here and return LSM_ERR_INVALID_ARGUMENT?